Extensions/xkit_preferences.js

```diff
@@ -2026,28 +2106,138 @@ XKit.extensions.xkit_preferences = new Object({
+			}
+			if (XKit.tools.get_setting("xkit_show_safari_news","false") === "true") {
+				$("#xkit-panel-enable-safari-news").addClass("selected");
+			}
+		} else if(XKit.browser().chrome) {
+			if (XKit.tools.get_setting("xkit_show_firefox_news","false") === "true") {
+				$("#xkit-panel-enable-firefox-news").addClass("selected");
+			}
+			if (XKit.tools.get_setting("xkit_show_chrome_news","true") === "true") {
+				$("#xkit-panel-enable-chrome-news").addClass("selected");
+			}
+			if (XKit.tools.get_setting("xkit_show_safari_news","false") === "true") {
+				$("#xkit-panel-enable-safari-news").addClass("selected");
+			}
+		} else if(XKit.browser().safari) {
+			if (XKit.tools.get_setting("xkit_show_firefox_news","false") === "true") {
+				$("#xkit-panel-enable-firefox-news").addClass("selected");
+			}
+			if (XKit.tools.get_setting("xkit_show_chrome_news","false") === "true") {
+				$("#xkit-panel-enable-chrome-news").addClass("selected");
+			}
+			if (XKit.tools.get_setting("xkit_show_safari_news","true") === "true") {
+				$("#xkit-panel-enable-safari-news").addClass("selected");
+			}
+		}
+		if (XKit.tools.get_setting("xkit_show_information","true") === "true") {
+			$("#xkit-panel-enable-information").addClass("selected");
+		}
+		if (XKit.tools.get_setting("xkit_show_extension_updates","true") === "true") {
+			$("#xkit-panel-enable-extension-updates").addClass("selected");
+		}
+		if (XKit.tools.get_setting("xkit_show_weekly_news","true") === "true") {
+			$("#xkit-panel-enable-weekly").addClass("selected");
+		}
+		if (XKit.tools.get_setting("xkit_show_developers","false") === "true") {
+			$("#xkit-panel-enable-developers").addClass("selected");
 		}
 
-		$("#xkit-panel-enable-feature-updates").click(function() {
-
-			if (XKit.tools.get_setting("xkit_show_feature_updates","true") === "true") {
-				$("#xkit-panel-enable-feature-updates").removeClass("selected");
-				XKit.extensions.xkit_preferences.news.mark_all_as_read();
-				XKit.tools.set_setting("xkit_show_feature_updates","false");
+		$("#xkit-panel-enable-firefox-news").click(function() {
+			var channel_setting = "xkit_show_firefox_news";
+			if (XKit.tools.get_setting(channel_setting,"true") === "true") {
+				$(this).removeClass("selected");
+				XKit.tools.set_setting(channel_setting,"false");
 			} else {
-				$("#xkit-panel-enable-feature-updates").addClass("selected");
-				XKit.tools.set_setting("xkit_show_feature_updates","true");
+				$(this).addClass("selected");
+				XKit.tools.set_setting(channel_setting,"true");
+			}
+		});
+		$("#xkit-panel-enable-chrome-news").click(function() {
+			var channel_setting = "xkit_show_chrome_news";
+			if (XKit.tools.get_setting(channel_setting,"true") === "true") {
+				$(this).removeClass("selected");
+				XKit.tools.set_setting(channel_setting,"false");
+			} else {
+				$(this).addClass("selected");
+				XKit.tools.set_setting(channel_setting,"true");
+			}
+		});
```

Comment:
For `#xkit-panel-enable-chrome-news` and the other selectors below, the functions passed into the click handlers are nearly identical. We should be able to factor them out into a separate method like:

```js
function toggle_setting_click_handler(setting) {
    return function() {
        if (XKit.tools.get_setting(setting, "true") === "true") {
            $(this).removeClass("selected");
            XKit.tools.set_setting(setting, "false");
        } else {
            $(this).addClass("selected");
            XKit.tools.set_setting(setting, "true");
        }
    };
}
```

And then we can do

```js
$("#xkit-panel-enable-firefox-news").click(toggle_setting_click_handler("xkit_show_firefox_news"));
$("#xkit-panel-enable-chrome-news").click(toggle_setting_click_handler("xkit_show_chrome_news"));
$("#xkit-panel-enable-safari-news").click(toggle_setting_click_handler("xkit_show_safari_news"));
```